src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -177,7 +179,18 @@ export class CommandDetectionCapability implements ICommandDetectionCapability {
 		// executed markers. This is done on command finished just in case command executed never
 		// happens (for example PSReadLine tab completion)
 		if (this._isWindowsPty) {
-			this._commandMarkers = this._commandMarkers.sort((a, b) => a.line - b.line);
+			if (this._commandMarkers.length === 0) {
+				// If the command start timeout doesn't happen before command finished, just use the
+				// current marker.
+				if (!this._currentCommand.commandStartMarker) {
```

Comment:
since the timeout is 0, when will this happen?

Reply:
It may happen when running a command while another is running, I don't think it does in practice currently due to event loop related things in node-pty/conpty where data but it might and could in the future.